jablib/src/test/java/org/jabref/logic/bibtex/BibEntryWriterTest.java

```diff
@@ -847,4 +849,31 @@ private BibEntry firstEntryFrom(final String bibContentText) throws JabRefExcept
                 .getEntries()
                 .getFirst();
     }
+
+    @ADR(49)
+    @Test
+    void lowercaseStandardAndPreserveCustomCasing() throws Exception {
+        String bibtexEntry = """
+                @Article{test,
+                  Author                   = {Foo Bar},
+                  Title                    = {My title},
+                  CustomField              = {Some value}
+                }
+                """.replace("\n", OS.NEWLINE);
+
+        BibEntry entry = firstEntryFrom(bibtexEntry);
+        entry.setField(new UnknownField("CustomField"), "Some other value");
```

Comment:
```suggestion
        BibEntry entry = firstEntryFrom(bibtexEntry)
            .withField(new UnknownField("CustomField"), "Some other value");
```

Reply:
For consistency reasons with the other tests better not.